src/test/scala/chiselTests/experimental/hierarchy/SeparateElaborationSpec.scala

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests.experimental.hierarchy
+
+import chiselTests.ChiselFunSpec
+import chisel3._
+import chisel3.stage.{ChiselGeneratorAnnotation, ChiselStage, DesignAnnotation}
+import chisel3.internal.ImportedDefinitionAnnotation
+import chisel3.experimental.hierarchy.{Definition, Instance}
+import firrtl.options.TargetDirAnnotation
+
+import java.nio.file.Paths
+import scala.io.Source
+
+class SeparateElaborationSpec extends ChiselFunSpec with Utils {
```

Comment:
can you add another test for the name conflict case? Where you have Definition(AddOne) and then in the second elaboration you also make a `newDef = Definition(AddOne)` + `Instance(newDef)`?